### PR TITLE
style: Refactor get_prompt function for readability and maintainability

### PR DESCRIPTION
**Pull request by Einswilli**
Optimize the get_prompt function by using a dictionary to map ActionType to hints, thus avoiding repetitions and making the code more readable and maintainable.

### CODE REVIEW

**Review by neubig — APPROVED** [review]
This looks cleaner, thanks!

File: agenthub/planner_agent/prompt.py [hunk 1]
```diff
@@ -149,62 +167,36 @@ def get_prompt(plan: Plan, history: List[Tuple[Action, Observation]]) -> str:
     history_dicts = []
     latest_action: Action = NullAction()
     for action, observation in sub_history:
+        history_dicts.append(action.to_dict())
+        if (
+            not isinstance(observation, NullObservation)
+            and "extras" in observation_dict
+            and "screenshot" in observation_dict["extras"]
+        ):
+            del observation_dict["extras"]["screenshot"]
```
Comment (rbren) [inline agenthub/planner_agent/prompt.py:176]: this got a little mixed up. We should only `append` if it's not a NullObservation, and we don't need to check NullObservation when deleting screenshot

Comment (Einswilli) [reply agenthub/planner_agent/prompt.py]: Thank you for the feedback! I've revised the get_prompt method to address the issues you raised. Now, the method only appends to history_dicts if it's not a NullAction, and I've removed the unnecessary check for NullObservation when deleting the screenshot. These changes should ensure that the method behaves as intended. Let me know if there's anything else that needs to be adjusted.

**Review by rbren — APPROVED** [review]

**Comment by rbren** [comment]
@Einswilli just a few lint issues

**Comment by rbren** [comment]
@Einswilli can you take a look at the lint issues?

**Comment by rbren** [comment]
Sorry this is such a whack-a-mole 🙃  feel free to ping me in slack if you need another look

You can also run the pre-commit command from the github action to lint it on your local machine

**Pull request merged**